micro-rdk/src/esp32/camera.rs

```diff
@@ -112,6 +142,38 @@ impl Camera for Esp32Camera {
             self.return_cam_frame(Some(ptr));
             return Ok(buffer);
         }
-        Err(CameraError::CameraCouldntGetFrame)
+        Err(CameraError::FailedToGetImage)
+    }
+
+    fn render_frame(&mut self, mut buffer: BytesMut) -> Result<BytesMut, CameraError> {
+        if let Some(ptr) = self.get_cam_frame() {
```

Comment:
If you don't want to create an owner type for the frame, you could at least use `defer!` here to ensure cleanup. Above in `get_image` too